pkg/controllers/user/controller.go

```diff
@@ -325,6 +336,27 @@ func (h *handler) removeKubeconfig(user *klum.User) error {
 	return nil
 }
 
+func (h *handler) OnKubeconfigChange(s string, kubeconfig *klum.Kubeconfig) (*klum.Kubeconfig, error) {
+	if kubeconfig == nil {
+		return nil, nil
+	}
+	user, err := h.kuser.Get(s, metav1.GetOptions{})
+	if err != nil {
+		log.Error(err)
+		return kubeconfig, nil
+	}
+
+	if h.cfg.GithubToken != "" {
+		err := github.UploadGithubSecret(kubeconfig, user, h.cfg.GithubURL, h.cfg.GithubToken)
+		if err != nil {
+			// Returning as if everything went well because we don't want to interfere with the kubeconfig creation
+			log.Error(err)
+			return kubeconfig, nil
```

Comment:
What happens if it fails? What's the retry policy? How could I force a retry?

Reply:
I wanted to introduce metrics in the near future to deal with figuring out when something like this happens.
If we fail the creation here, credentials are not created at all and get rescheduled.
Maybe I thought this feature too loosely coupled to the main objective of this controller.

Reply:
With the changes in [62d0142](https://github.com/jadolg/klum/pull/19/commits/62d014283a26312ea9998ea8d87c3283f759cd81) now if it's supposed to create a secret and it fails then it errors.